src/regex/pcre2.cr

```diff
@@ -123,9 +137,24 @@ module Regex::PCRE2
     LibPCRE2.general_context_create(->(size : LibC::Int, data : Void*) { GC.malloc(size) }.pointer, ->(pointer : Void*, data : Void*) { GC.free(pointer) }.pointer, nil)
   end
 
+  # Returns a JIT stack that's shared in the current thread.
+  #
+  # Only a single `match` function can run per thread at any given time, so there
+  # can't be any concurrent access to the JIT stack.
+  @[ThreadLocal]
+  class_getter jit_stack : LibPCRE2::JITStack do
+    jit_stack = LibPCRE2.jit_stack_create(32_768, 1_048_576, Regex::PCRE2.general_context)
```

Comment:
Maybe for a follow up, but these magic numbers could be or'ed with the `ENV` in order to have some control. I'm thinking in particular of memory-restricted apps.